modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -32,6 +32,9 @@ public function getInfo() {
     return [
       // The order to which the coupon will be applied to.
       '#order' => NULL,
+      '#code_label' => t('Coupon code'),
+      '#code_description' => t('Enter your coupon code here.'),
+      '#apply_label' => t('Apply coupon'),
```

Comment:
>  main title and description.

What do you mean? I imagined the checkout pane would allow for people to customize this default text, such as remove or sub out "coupon" for just "promotion".

When implementing the element I immediately needed to extend the class to make my own element in order to meet client requirements. Which if I had to on first go, I'm assuming 80% of all of our contributors would have to. And most will use a form_alter instead of extending element. I see this as a way to make cleaner code for end users (because checkout pane can provide a UI.)